libs/common/src/helpers/string.ts

```diff
@@ -1,16 +1,104 @@
 /**
- * Checks whether the provided value is a `string`
+ * Checks if a value is a string.
+ *
+ * @param value - The value to check.
+ * @returns True if the value is a string, otherwise false.
  */
 export const isString = (value: unknown): value is string => typeof value === 'string' || value instanceof String;
 
 /**
- *  Checks whether the provided value is `null`, `undefined`, or an empty
- *  (0 length, or whitespace only) `string`.
+ * Checks if the given value is null, undefined, or an empty string.
+ *
+ * @param value - The value to check.
+ * @returns `true` if the value is null, undefined, or an empty string; otherwise, `false`.
  */
 export const isNullUndefinedOrEmptyString = (value: unknown): boolean => !value || (isString(value) && !value.trim().length);
 
 /**
- * Checks whether the provided value is non-empty (non-0 length and
- * non-whitespace only) `string`.
+ * Checks if the given value is a non-empty string.
+ *
+ * This function first verifies if the value is a string using the `isString` function,
+ * and then ensures that the string is neither null, undefined, nor empty using the
+ * `isNullUndefinedOrEmptyString` function.
+ *
+ * @param value - The value to check.
+ * @returns True if the value is a non-empty string, otherwise false.
  */
 export const isNonEmptyString = (value: unknown): value is string => isString(value) && !isNullUndefinedOrEmptyString(value);
+
+/**
+ * Replaces all special characters in a given string with a specified replacement string.
+ * Optionally preserves the file extension if specified.
+ *
+ * @param {string} value - The input string to be sanitized.
+ * @param {string} [replace=''] - The string to replace special characters with. Defaults to an empty string.
+ * @param {boolean} [fileExtension=false] - Whether to preserve the file extension. Defaults to false.
+ * @returns {string} - The sanitized string with special characters replaced.
```

Comment:
not a blocker, we do not need the types here in the params

Reply:
Will remove all the params from TSDocs in one single PR across the whole repo.

Reply:
Will remove all the params from TSDocs in one single PR across the whole repo.